filters/include/pcl/filters/spherical_voxel_grid.h

```diff
@@ -0,0 +1,297 @@
+      }
+
+      /** \brief Get the field filter limits (min/max) set by the user. The default values are -FLT_MAX, FLT_MAX.
+        * \param[out] limit_min the minimum allowed field value
+        * \param[out] limit_max the maximum allowed field value
+        */
+      inline void
+      getFilterLimits (double &limit_min, double &limit_max) const
+      {
+        limit_min = filter_limit_min_;
+        limit_max = filter_limit_max_;
+      }
+
+      /** \brief Set to true if we want to return the data outside the interval specified by setFilterLimits (min, max).
+        * Default: false.
+        * \param[in] limit_negative return data inside the interval (false) or outside (true)
+        */
+      inline void
+      setFilterLimitsNegative (const bool limit_negative) { filter_limit_negative_ = limit_negative; }
+
+      /** \brief Get whether the data outside the interval (min/max) is to be returned (true) or inside (false).
+        * \param[out] limit_negative true if data \b outside the interval [min; max] is to be returned, false otherwise
+        */
+      inline void
+      getFilterLimitsNegative (bool &limit_negative) const { limit_negative = filter_limit_negative_; }
+
+      /** \brief Get whether the data outside the interval (min/max) is to be returned (true) or inside (false).
+        * \return true if data \b outside the interval [min; max] is to be returned, false otherwise
+        */
+      inline bool
+      getFilterLimitsNegative () const { return (filter_limit_negative_); }
+
+  protected:
+      /** \brief The radial size of the voxels. */
+      float leaf_size_r_;
+
+      /** \brief The number of radial divisions. */
+      int leaf_r_divisions_;
+
+      /** \brief The number of vertical divisions. */
+      int leaf_theta_divisions_;
+
+      /** \brief The number of horizontal divisions. */
+      int leaf_phi_divisions_;
+
+      /** \brief The size of the vertical divisions (radians). */
+      float leaf_size_theta_;
+
+      /** \brief The size of the horizontal divisions (radians). */
+      float leaf_size_phi_;
+
+      /** \brief Set to true if all fields need to be downsampled, or false if just XYZ. */
+      bool downsample_all_data_;
+
+      /** \brief Minimum number of points per voxel for the centroid to be computed */
+      unsigned int min_points_per_voxel_;
+
+      /** \brief The desired user filter field name. */
+      std::string filter_field_name_;
+
+      /** \brief The minimum allowed filter value a point will be considered from. */
+      double filter_limit_min_;
+
+      /** \brief The maximum allowed filter value a point will be considered from. */
+      double filter_limit_max_;
+
+      /** \brief Set to true if we want to return the data outside (\a filter_limit_min_;\a filter_limit_max_). Default: false. */
+      bool filter_limit_negative_;
+
+      /** Cartesian origin of the spherical coordinate system. */
+      Eigen::Vector4f filter_origin_;
```

Comment:
Since you store and eigen vectorizable type internally in your class you need to define the [Eigen macro](https://eigen.tuxfamily.org/dox/group__TopicStructHavingEigenMembers.html) for overloading the new operator.

Reply:
Are you sure that's enough? I never looked into the internals of the macro to understand what exactly it does.

Reply:
https://forum.kde.org/viewtopic.php?f=74&t=124742#p328875